src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +129,69 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/32316", TestRuntimes.Mono)]
+        public static void ZipArchiveEntry_AppContext_Switch_SuppressHeaderValidation(bool suppressValidation)
+        {
+            RemoteExecutor.Invoke((suppressString) =>
+            {
+                bool suppress = bool.Parse(suppressString);
+                MemoryStream stream = populateStream().Result;
+                PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+                using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+                {
+                    // Set the AppContext Switch
+                    AppContext.SetSwitch("System.Compression.ZipArchiveEntry.SuppressHeaderValidation", suppress);
```

Comment:
Be careful about this.  Since you read and cache the value of the AppContext switch in a static you are sensitive to where that static runs.  To be safe from JIT changes (like you may be seeing on mono) you might want to set this in one method, then put the rest of your test method in a non-inlineable method.  This could even be an overload to RemoteExecutor.Invoke, like InvokeWithAppContextSwitch.